src/Jitter_CodeGen_x86_Fpu_Avx.cpp

```diff
@@ -8,23 +8,61 @@ void CCodeGen_x86::Emit_Fpu_Avx_MemMem(const STATEMENT& statement)
 	auto dst = statement.dst->GetSymbol().get();
 	auto src1 = statement.src1->GetSymbol().get();
 
-	((m_assembler).*(FPUOP::OpEdAvx()))(CX86Assembler::xMM0, CX86Assembler::xMM0, MakeMemoryFpSingleSymbolAddress(src1));
-	m_assembler.VmovssEd(MakeMemoryFpSingleSymbolAddress(dst), CX86Assembler::xMM0);
+	auto dstRegister = PrepareSymbolRegisterDefFpu(dst, CX86Assembler::xMM0);
+	auto src1Register = PrepareSymbolRegisterUseFpuAvx(src1, CX86Assembler::xMM1);
+
+	((m_assembler).*(FPUOP::OpEdAvx()))(dstRegister, dstRegister, CX86Assembler::MakeXmmRegisterAddress(src1Register));
+
+	CommitSymbolRegisterFpuAvx(dst, dstRegister);
+
+}
+
+void CCodeGen_x86::Emit_Fp_Avx_Neg_VarVar(const STATEMENT& statement)
+{
+	auto dst = statement.dst->GetSymbol().get();
+	auto src1 = statement.src1->GetSymbol().get();
+
+	auto tmpIntRegister = CX86Assembler::rAX;
+	auto tmpXMMRegister = CX86Assembler::xMM2;
+	auto tmpXMM2Register = CX86Assembler::xMM3;
+	auto dstRegister = PrepareSymbolRegisterDefFpu(dst, CX86Assembler::xMM0);
+	auto src1Register = PrepareSymbolRegisterUseFpuAvx(src1, CX86Assembler::xMM1);
+
+	m_assembler.MovId(tmpIntRegister, 0x80000000);
```

Comment:
We can avoid crossing register domains here if we use this sequence:

vpcmpeqd xmm0, xmm0, xmm0
vpslld xmm0, xmm0, 31

You will end up with 0x80000000 in xmm0 with that.